app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt

```diff
@@ -127,12 +127,14 @@ class BrowserFragment : IWebViewLifecycleFragment() {
                         NavigationEvent.VAL_CHECKED -> {
                             CustomTilesManager.getInstance(context).pinSite(context, url,
                                     webView?.takeScreenshot())
+                            browserOverlay.addPinnedSiteToTiles()
                             showCenteredTopToast(context, R.string.notification_pinned_site, 0, TOAST_Y_OFFSET)
                         }
                         NavigationEvent.VAL_UNCHECKED -> {
                             url.toUri()?.let {
-                                if (BundledTilesManager.getInstance(context).unpinSite(context, it)
-                                        || CustomTilesManager.getInstance(context).unpinSite(context, url)) {
+                                val tileId = BundledTilesManager.getInstance(context).unpinSite(context, it) ?: CustomTilesManager.getInstance(context).unpinSite(context, url)
+                                if (!tileId.isNullOrEmpty()) {
+                                    browserOverlay.removePinnedSiteFromTiles(tileId!!)
```

Comment:
Okay, red flag, `!!` is not okay. Try to use `!!` it as little as possible, basically, only when there isn't another way and you actually **want** it to crash if it's null. (e.g. you're in some really unexpected state, and you'd rather crash so you can get a crash report, or whatever)

In this case, you actually know it's not null, so (I learned by asking mcomella) if you do a null check, Kotlin will cast it for you. Since you want to also do the empty check, you should do something like
`if (tileId != null && !tileId.isEmpty())`